### PR TITLE
ExecutorConfigurationSupport to allow millisecond precision for await termination period

### PR DESCRIPTION
**Pull request by ttddyy**
I noticed `ExecutorConfigurationSupport`(_`ThreadPoolTask[Executor|Scheduler]`_) only takes seconds for await termination.

I'm writing a graceful shutdown logic for k8s environment and awaiting by second is a bit large granularity to control the shutdown/await. Graceful shutdown get triggered by liveness probe and the frequency for liveness probe is not so long. 

This PR changes the minimum unit to milliseconds and adds a method to take `Duration` to specify the await termination.

### CODE REVIEW

File: spring-context/src/main/java/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.java [hunk 1]
```diff
@@ -145,7 +146,16 @@ public void setWaitForTasksToCompleteOnShutdown(boolean waitForJobsToCompleteOnS
...
+		this.awaitTerminationDuration = Duration.ofSeconds(awaitTerminationSeconds);
+	}
+
+	/**
+	 *
+	 * @param awaitTerminationDuration time to await to shutdown internal executor
+	 * @since 5.2.4
+	 */
+	public void setAwaitTerminationDuration(Duration awaitTerminationDuration) {
+		this.awaitTerminationDuration = awaitTerminationDuration;
```
Comment (rstoyanchev) [inline spring-context/src/main/java/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.java]: I think should be called just `setAwaitTermination`.

Comment (rstoyanchev) [inline spring-context/src/main/java/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.java]: I think should be called just `setAwaitTermination`.

File: spring-context/src/main/java/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.java [hunk 2]
```diff
@@ -145,7 +146,16 @@ public void setWaitForTasksToCompleteOnShutdown(boolean waitForJobsToCompleteOnS
 	 * @see java.util.concurrent.ExecutorService#awaitTermination
 	 */
 	public void setAwaitTerminationSeconds(int awaitTerminationSeconds) {
-		this.awaitTerminationSeconds = awaitTerminationSeconds;
+		this.awaitTerminationDuration = Duration.ofSeconds(awaitTerminationSeconds);
+	}
+
+	/**
+	 *
```
Comment (rstoyanchev) [inline spring-context/src/main/java/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.java]: Add Javadoc (won't pass checkstyle).

**Review by rstoyanchev — CHANGES_REQUESTED** [review]

**Comment by rstoyanchev** [comment]
I've turned that into `awaitTerminationMillis` which is usable for XML config, consistent with the rest of the framework, itself a shortcut for a Duration.

**Comment by ttddyy** [comment]
cool, thanks!!

**Pull request closed without merging**